### ISSUE TITLE
Windows Compilation Error

### ISSUE THREAD

**Issue by gmoore016** [Yale-Medicaid/sparsepolicytree#5]
**Describe the bug**
When I use `devtools::install_github` to install your package, I get the following terminal output: 
```
devtools::install_github("Yale-Medicaid/sparsepolicytree")
Downloading GitHub repo Yale-Medicaid/sparsepolicytree@HEAD
── R CMD build ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file 'C:\Users\gmoor\AppData\Local\Temp\RtmpGwuz2P\remotes4e7c779a4f87\Yale-Medicaid-sparsepolicytree-682e472/DESCRIPTION'
─  preparing 'sparsepolicytree':
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building 'sparsepolicytree_0.0.0.9000.tar.gz'
   
Installing package into 'C:/Users/gmoor/AppData/Local/R/win-library/4.3'
(as 'lib' is unspecified)
* installing *source* package 'sparsepolicytree' ...
** using staged installation
** libs
using C compiler: 'gcc.exe (GCC) 12.2.0'
rm -Rf sparsepolicytree.dll ./rust/target/x86_64-pc-windows-gnu/release/libsparsepolicytree.a entrypoint.o
gcc  -I"C:/PROGRA~1/R/R-43~1.1/include" -DNDEBUG     -I"C:/rtools43/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c entrypoint.c -o entrypoint.o
mkdir -p ./rust/target/libgcc_mock
cd ./rust/target/libgcc_mock && \
        touch gcc_mock.c && \
        gcc -c gcc_mock.c -o gcc_mock.o && \
        ar -r libgcc_eh.a gcc_mock.o && \
        cp libgcc_eh.a libgcc_s.a
C:\rtools43\x86_64-w64-mingw32.static.posix\bin\ar.exe: creating libgcc_eh.a
# CARGO_LINKER is provided in Makevars.ucrt for R >= 4.2
export CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER="x86_64-w64-mingw32.static.posix-gcc.exe" && \
        export LIBRARY_PATH="${LIBRARY_PATH};/c/Users/gmoor/AppData/Local/Temp/Rtmp6N8SkW/R.INSTALL481c4fe821ec/sparsepolicytree/src/./rust/target/libgcc_mock" && \
        cargo +stable-gnu build --target=x86_64-pc-windows-gnu --lib --release --manifest-path=./rust/Cargo.toml --target-dir ./rust/target
   Compiling autocfg v1.1.0
   Compiling winapi-x86_64-pc-windows-gnu v0.4.0
   Compiling winapi v0.3.9
   Compiling crossbeam-utils v0.8.13
   Compiling proc-macro2 v1.0.47
   Compiling cfg-if v1.0.0
   Compiling unicode-ident v1.0.5
   Compiling quote v1.0.21
   Compiling syn v1.0.103
   Compiling scopeguard v1.1.0
   Compiling rayon-core v1.10.1
   Compiling extendr-engine v0.3.1
   Compiling rawpointer v0.2.1                                                                                                                                                   
   Compiling num_cpus v1.14.0
   Compiling matrixmultiply v0.3.2                                                                                                                                               
   Compiling either v1.8.0                                                                                                                                                       
   Compiling extendr-api v0.3.1                                                                                                                                                  
   Compiling paste v1.0.9
   Compiling lazy_static v1.4.0
   Compiling itertools v0.10.5
   Compiling iter_utils v0.1.0
   Compiling num-traits v0.2.15                                                                                                                                                  
   Compiling memoffset v0.7.1                                                                                                                                                    
   Compiling crossbeam-epoch v0.9.12                                                                                                                                             
   Compiling num-integer v0.1.45                                                                                                                                                 
   Compiling binary_search_tree v0.2.2                                                                                                                                           
   Compiling crossbeam-channel v0.5.6
   Compiling crossbeam-deque v0.8.2
   Compiling libR-sys v0.3.0
   Compiling num-complex v0.4.2
   Compiling ordered-float v3.4.0
   Compiling rayon v1.6.0
   Compiling ndarray v0.15.6                                                                                                                                                     
   Compiling extendr-macros v0.3.1
   Compiling sparsepolicytree v0.1.0 (C:\Users\gmoor\AppData\Local\Temp\Rtmp6N8SkW\R.INSTALL481c4fe821ec\sparsepolicytree\src\rust)
    Finished release [optimized] target(s) in 25.75s
gcc -shared -s -static-libgcc -o sparsepolicytree.dll tmp.def entrypoint.o -L./rust/target/x86_64-pc-windows-gnu/release -lsparsepolicytree -lws2_32 -ladvapi32 -luserenv -lbcrypt -LC:/rtools43/x86_64-w64-mingw32.static.posix/lib/x64 -LC:/rtools43/x86_64-w64-mingw32.static.posix/lib -LC:/PROGRA~1/R/R-43~1.1/bin/x64 -lR
C:\rtools43\x86_64-w64-mingw32.static.posix\bin/ld.exe: ./rust/target/x86_64-pc-windows-gnu/release/libsparsepolicytree.a(std-ca5208825e97b4ba.std.687851ba-cgu.0.rcgu.o): in function `std::sys::windows::fs::open_link_no_reparse':
/rustc/90c541806f23a127002de5b4038be731ba1458ca/library\std\src\sys\windows/fs.rs:800: undefined reference to `NtCreateFile'
C:\rtools43\x86_64-w64-mingw32.static.posix\bin/ld.exe: /rustc/90c541806f23a127002de5b4038be731ba1458ca/library\std\src\sys\windows/fs.rs:829: undefined reference to `RtlNtStatusToDosError'
C:\rtools43\x86_64-w64-mingw32.static.posix\bin/ld.exe: ./rust/target/x86_64-pc-windows-gnu/release/libsparsepolicytree.a(std-ca5208825e97b4ba.std.687851ba-cgu.0.rcgu.o): in function `std::sys::windows::handle::Handle::synchronous_read':
/rustc/90c541806f23a127002de5b4038be731ba1458ca/library\std\src\sys\windows/handle.rs:241: undefined reference to `NtReadFile'
C:\rtools43\x86_64-w64-mingw32.static.posix\bin/ld.exe: /rustc/90c541806f23a127002de5b4038be731ba1458ca/library\std\src\sys\windows/handle.rs:272: undefined reference to `RtlNtStatusToDosError'
C:\rtools43\x86_64-w64-mingw32.static.posix\bin/ld.exe: ./rust/target/x86_64-pc-windows-gnu/release/libsparsepolicytree.a(std-ca5208825e97b4ba.std.687851ba-cgu.0.rcgu.o): in function `std::sys::windows::handle::Handle::synchronous_write':
/rustc/90c541806f23a127002de5b4038be731ba1458ca/library\std\src\sys\windows/handle.rs:290: undefined reference to `NtWriteFile'
C:\rtools43\x86_64-w64-mingw32.static.posix\bin/ld.exe: /rustc/90c541806f23a127002de5b4038be731ba1458ca/library\std\src\sys\windows/handle.rs:318: undefined reference to `RtlNtStatusToDosError'
collect2.exe: error: ld returned 1 exit status
no DLL was created
ERROR: compilation failed for package 'sparsepolicytree'
* removing 'C:/Users/gmoor/AppData/Local/R/win-library/4.3/sparsepolicytree'
Warning message:
In i.p(...) :
  installation of package 'C:/Users/gmoor/AppData/Local/Temp/RtmpGwuz2P/file4e7c16885b1b/sparsepolicytree_0.0.0.9000.tar.gz' had non-zero exit status
```

**To Reproduce**
1. Install R 4.3.1
2. Install rustc 1.70.0 using rustup 1.26.0
3. Install devtools 2.4.5
4. Run `devtools::install_github("Yale-Medicaid/sparsepolicytree")`

**Expected behavior**
It should install the `sparsepolicytree` package

**Desktop:**
 - OS: Windows 11
 - Version: 22H2

**Additional context**
This issue seems to match the symptoms documented [here](https://yutani.rbind.io/post/rust-1.70-and-build-failure-on-windows/), which makes me believe it's an issue with the package rather than an issue with my environment.

**Comment by beniaminogreen** [comment]
Hi,  thanks for leaving such a detailed issue request, and for providing that link! Sorry it's taken such a long time for me to respond to this - for some reason I did not receive the email notification for this thread. 

You were completely right about the cause of the issue, and I merged in the fix described in that article in #6. Just double checked on our Windows server and the package is compiling again now, but please let me know if this does not solve the issue.

**Comment by beniaminogreen** [comment]
I am closing this as I think it was resolved in #6. Happy to re-open if there is still and issue.
